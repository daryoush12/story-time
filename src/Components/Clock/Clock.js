import React, { useState } from 'react'

export default function Clock() {
    const [count, setCount] = useState(0)

    const Increase = (e) => {
        e.preventDefault()
        setCount(count + 1)
    }

    return (
        <div>
            <p>{count}</p>
            <button onClick={Increase}>Increase</button>
        </div>
    )
}
