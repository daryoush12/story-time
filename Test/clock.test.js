// Link.react.test.js
import React from 'react'
import renderer from 'react-test-renderer'
import Clock from '../src/Components/Clock/Clock.js'

test('Clock count changes when button is clicked', () => {
    const component = renderer.create(<Clock />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.find('button').simulate()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
