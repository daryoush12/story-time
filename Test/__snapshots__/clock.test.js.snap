// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock count changes when button is clicked 1`] = `
<div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;

exports[`Clock count changes when button is clicked 2`] = `
<div>
  <p>
    1
  </p>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;